sentinelhub/constants.py

```diff
@@ -111,7 +112,41 @@ class _OrbitDirection(Enum):
     BOTH = 'both'
 
 
-class DataSource(Enum):
+class DataSourceMeta(EnumMeta):
+    """
+    Meta class used to add custom (BYOC) datasources to DataSource enumerator.
+
+    BYOC stands for Bring Your Own Data. For more details see: https://docs.sentinel-hub.com/api/latest/#/API/byoc
+    """
+    def __call__(cls, collection_id, *args, **kwargs):
```

Comment:
Here you assume that `collection_id` will always be string. Because of that things like `DataSource(DataSource.SENTINEL2_L1C)` or `DataSource(DataSource.SENTINEL2_L1C.value)` don't work anymore.

I recommend adding a check at the beginning of this method
```Python
if not isinstance(collection_id, str):
    return super().__call__(collection_id, *args, **kwargs)
```